sycl/doc/extensions/proposed/sycl_ext_oneapi_group_load_store_extended.asciidoc

```diff
@@ -0,0 +1,223 @@
+= sycl_ext_oneapi_group_load_store
+
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+:dpcpp: pass:[DPC++]
+:language: {basebackend@docbook:c++:cpp}
+
+== Introduction
+
+IMPORTANT: This specification is a draft.
+
+
+== Notice
+
+[%hardbreaks]
+Copyright (c) 2022 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+== Dependencies
+
+This extension is written against the SYCL 2020 specification, Revision 6.
+All references below to the "core SYCL specification" or to section numbers in
+the SYCL specification refer to that revision.
+
+This extension also depends on the following other SYCL extensions:
+
+* link:https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc[
+  sycl_ext_oneapi_group_sort] (`group_with_scratchpad` class used as GroupHelper).
+
+* link:../experimental/sycl_ext_oneapi_properties.asciidoc[
+  sycl_ext_oneapi_properties].
+
+== Overview
+
+This extention defines free functions for load/store operations within work
+group scope.
+
+== Single Value Functions
+
+[source,c++]
+----
+// Load API
+template<typename Group, typename InputIteratorT,
+         typename OutputT, typename Properties = properties<>>
+void joint_load(Group gh, InputIteratorT in_ptr,
+                      OutputT& out, Properties = {});
+
+// Store API
+template<typename Group, typename InputT,
+         typename OutputIteratorT, typename Properties = properties<>>
+void joint_store(Group gh, InputT in,
+                      OutputIteratorT out_ptr, Properties = {});
+----
+
```

Comment:
I think we need to add the preconditions etc from below to these functions too.  This should be straightforward, mostly a copy-paste.

Reply:
@Pennycook, thank you, added. Please, also check again for span-based API, I changed the wording for _Constraints_ section, as the GroupHelper template parameter can be just a group as well, as it is shown in Example 1.

Also, I fixed the single value store function with const InputT& instead of just InputT.

@dkhaldi, @Pennycook, I also added an overload for sycl::vec (new functions (3) and (4)), as in the span-based case it looks a bit ugly: https://github.com/intel/llvm-test-suite/blob/0fd0b218b4e0a4c2b486f3863b0244f54aa0717d/SYCL/GroupLoadStore/group_load_store_span_api.cpp#L167

unlike marray, which is nice with begin() and end(): https://github.com/intel/llvm-test-suite/blob/0fd0b218b4e0a4c2b486f3863b0244f54aa0717d/SYCL/GroupLoadStore/group_load_store_span_api.cpp#L136

please, advise if we should keep it

Reply:
@aelizaro, thank you. These are good examples to show this feature works on vec and marray.
- Is there an another document that already talks about construction of span using vec and marray? 
- Does the example you added to llvm-test-suite currently run on CPU and GPU? I did not see XFAIL there.

Reply:
@dkhaldi, thank you

* For the construction of span using vec and marray I don't know if there are any specific docs, but they both are guaranteed to be in memory sequentially and contiguously: https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#memory-layout-and-alignment
* I assume examples shouldn't run now at all, since for this functionality there is no implementation available yet, only API proposal and initial prototypes